tests/version.go

```diff
@@ -0,0 +1,49 @@
+package tests
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/moby/buildkit/util/testutil/integration"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/mod/module"
+	"golang.org/x/mod/semver"
+)
+
+var versionTests = []func(t *testing.T, sb integration.Sandbox){
+	testVersion,
+}
+
+func testVersion(t *testing.T, sb integration.Sandbox) {
+	cmd := buildxCmd(sb, withArgs("version"))
+	out, err := cmd.CombinedOutput()
+	require.NoError(t, err, string(out))
+
+	// Output should be a single line.
+	version := string(out)
+	require.Equal(t, 1, strings.Count(version, "\n"))
+
+	// Split by spaces into three fields.
+	components := strings.Split(version, " ")
+	require.Equal(t, 3, len(components))
+
+	// First component should be the package clause.
+	// This can be any valid package path for Go
+	// so don't set too many restrictions here.
+	// Just checking if the import path is a valid Go
+	// path should be suitable enough to make sure this is ok.
+	// Using CheckImportPath instead of CheckPath as it is less
+	// restrictive.
+	require.NoError(t, module.CheckImportPath(components[0]), "First component was not a valid import path: %+v", components[0])
+
+	// Second component should be a version.
+	// This defaults to something that's still compatible
+	// with semver.
+	require.True(t, semver.IsValid(components[1]), "Second component was not valid semver: %+v", components[1])
```

Comment:
It seems to make sense that version is not semver when running tests. We use `git describe` to set the version: https://github.com/docker/buildx/blob/c010d3de8dc8771f89b0ec978bb58105652071d0/hack/git-meta#L7

And looking at the GHA workflow: https://github.com/docker/buildx/blob/c010d3de8dc8771f89b0ec978bb58105652071d0/.github/workflows/build.yml#L78-L80

We don't fetch tags so only commit sha will be returned. It should work if you set:

```yaml
      -
        name: Checkout
        uses: actions/checkout@v3
        with:
          fetch-depth: 0
```

See https://github.com/actions/checkout/#usage

> ```yaml
> # Number of commits to fetch. 0 indicates all history for all branches and tags.
> # Default: 1
> fetch-depth: ''
> ```